dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/trackedentity/DefaultTrackedEntityService.java

```diff
@@ -341,12 +340,6 @@ public TrackedEntities getTrackedEntities(TrackedEntityOperationParams operation
   }
 
   public List<Long> getTrackedEntityIds(TrackedEntityQueryParams params) {
-    if (!params.hasProgram()) {
```

Comment:
@stian-sandvold @Bekkalizer @JoakimSM that check is implementing a logic that is not documented in any place and it is potentially harmful in the sense that some TE will not be retrieved based on a filter that is not specified by the client.

If you cannot see a strong reason to maintain this here, I am happy to remove it

Reply:
I am not really sure that this code is doing this kind of validation.
What this code is doing is adding a filter based on a flag present in the attribute.
This means that if there are 2 attributes, A and B where A has `DisplayInListNoProgram` flag as true and B has `DisplayInListNoProgram` as false, then if a client is not providing a program, only the TEs that have a value for attribute A will be returned.

I am not sure how `DisplayInListNoProgram` is set and exactly what it means. But anyway this check here is not preventing searching on any attribute or filtering the attributes that are returned.